DigitalLearningSolutions.Web/ViewModels/TrackingSystem/Delegates/DelegateGroups/DelegateGroupsViewModel.cs

```diff
@@ -2,15 +2,59 @@
 {
     using System.Collections.Generic;
     using System.Linq;
+    using DigitalLearningSolutions.Data.Models.CustomPrompts;
     using DigitalLearningSolutions.Data.Models.DelegateGroups;
+    using DigitalLearningSolutions.Web.Helpers;
+    using DigitalLearningSolutions.Web.ViewModels.Common.SearchablePage;
 
-    public class DelegateGroupsViewModel
+    public class DelegateGroupsViewModel : BaseSearchablePageViewModel
     {
-        public DelegateGroupsViewModel(IEnumerable<Group> groups)
+        public DelegateGroupsViewModel(
+            IEnumerable<Group> groups,
+            IEnumerable<CustomPrompt> registrationPrompts,
+            string sortBy,
+            string sortDirection,
+            string? filterBy,
+            int page
+        ) : base(null, page, true, sortBy, sortDirection, filterBy)
         {
-            DelegateGroups = groups.Select(g => new SearchableDelegateGroupViewModel(g));
+            groups = groups.ToList();
+            var sortedItems = GenericSortingHelper.SortAllItems(
+                groups.AsQueryable(),
+                sortBy,
+                sortDirection
+            );
+            var filteredItems = FilteringHelper.FilterItems(sortedItems.AsQueryable(), filterBy).ToList();
+            MatchingSearchResults = filteredItems.Count;
+            SetTotalPages();
+            var paginatedItems = GetItemsOnCurrentPage(filteredItems);
+

```

Comment:
Spacing can often be used to separate code into small logical sections when separating things into functions is not justified. In this case, the "paginatedItems" declaration seems like something that's unrelated to the code above it and more to do with the code below it. (I might be missing something, but the spacing here is making be a little bit confused)